pallene/checker.lua

```diff
@@ -838,7 +838,7 @@ function Checker:check_exp_synthesize(exp)
         type_error(exp.loc, "missing type hint for initializer")
 
     elseif tag == "ast.Exp.Lambda" then
-        type_error(exp.loc, "missing type hint for lambda")
+        type_error(exp.loc, "higher order functions not implemented.")
```

Comment:
This "noy yet implemented" error should go into the `check_exp_verify`. 

For `check_exp_synthesize` the current error is correct. It is not possible to typecheck a lambda in isolation; we must always know the expected type first. (This is similar issue to what happens in table literals).